### ISSUE TITLE
vsphere_virtual_machine product_key setting documentation mismatch

### ISSUE THREAD

**Issue by sprokopiak** [hashicorp/terraform#6690]
There is a mismatch between the documentation in `vsphere_virtual_machine` for `windows_opt_config.product_key` and the implementation.  The documentation lists the setting as optional; however, if I have the `windows_opt_config` block in my tf without a `product_key` listed, I receive an error that the field is required.

Looking at the code, `product_key` is set as required.

I'm not sure which is correct. (I'm guessing optional and that the code is wrong).

### Terraform Version
v0.6.16

### Affected Resource(s)
Please list the resources as a list, for example:
- vsphere_virtual_machine

### Terraform Configuration Files
```hcl
    windows_opt_config {
        domain = "perqalab.local"
        domain_user = "Administrator"
        domain_user_password = "${var.admin_password}"
    }
```

### Debug Output
```
There are warnings and/or errors related to your configuration. Please
fix these before continuing.

Errors:

  * vsphere_virtual_machine.web: "windows_opt_config.0.product_key": required field is not set
```

### Panic Output
N/A

### Expected Behavior
Terraform should run without having to provide the `product_key` setting.

### Actual Behavior
Terraform throws an error about missing the required `product_key` setting.

### Steps to Reproduce
1. `terraform apply`

**Comment by chrislovecnm** [comment]
So you are wanting the `product_key` to be optional?  Sounds reasonable.

@jen20 @stack72 can we get a vsphere label on this :)

**Comment by sprokopiak** [comment]
@chrislovecnm If there's a reason why the `product_key` is required when specifying `windows_opt_config`, that's fine. But the documentation should reflect that the setting is required.

As it stands, [the documentation](https://www.terraform.io/docs/providers/vsphere/r/virtual_machine.html#product_key) states `product_key` is optional, yet [the code](https://github.com/hashicorp/terraform/blob/master/builtin/providers/vsphere/resource_vsphere_virtual_machine.go#L219) states it's required.

**Comment by chrislovecnm** [comment]
@sprokopiak I am not a windows guy at all... I don't think it is required, but I would not be able to test it. 

@thetuxkeeper @markpeek @dkalleg @kristinn any need to have `product_key` required?

Yes the documentation should be updated.  All PR's are welcome!!

**Comment by thetuxkeeper** [comment]
There are evaluation editions where you have some months until you have to enter the product key, so making it optional would be the right way.
But I never deployed a windows except my personal one, so I'm not sure if you need some evaluation key or if it's really completely without key.

**Comment by chrislovecnm** [comment]
Can any windows gurus help out here? @sprokopiak do you need a license key to deploy? Just in general.  This is not a TF specific issue. More windows issue.

**Comment by sprokopiak** [comment]
From what I can tell, our company machines do not need a license to deploy. They are usually the evaluation versions that @thetuxkeeper mentioned, with 180 days to evaluate.

During my testing so far, I was able to deploy a Windows VM on a domain by providing `product_key = ""`. Seemed to work fine. I've also deployed a Windows VM without providing the `windows_opt_config` at all, as I didn't have a need to set any of those configuration values.
